crates/autopilot/src/domain/settlement/observation.rs

```diff
@@ -0,0 +1,36 @@
+//! Aggregated type, based on the mined settlement transaction.
+//!
+//! It contains all important information about the mined settlement, including
+//! the surplus and fees.
+
+// TODO
+
+use {
+    crate::domain::{auction, eth, settlement},
+    std::collections::HashMap,
+};
+
+pub struct Observation {
+    auction: auction::Id,
+    surplus: settlement::NormalizedSurplus,
+    fees: settlement::Fees,
+}
+
+impl Observation {
+    /// Creates a new observation from the given settlement and prices.
+    ///
+    /// No error handling since observation is sometimes partial but still
+    /// useful. ?
+    pub fn new(
```

Comment:
Might also make sense to have this constructor be a method on `Encoded` as well.

Reply:
done